enterprise/api/v1/views.py

```diff
@@ -1483,3 +1484,56 @@ def update(self, request, *args, **kwargs):
     )
     def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
+
+    @action(methods=['post'], detail=True, url_path='link-user')
+    def link_user(self, request, pk=None):
+        """
+        Post
+        Links user using enterprise_customer_key
+        /enterprise/api/enterprise-customer-invite-key/{enterprise_customer_key}/link-user
+
+        Given a enterprise_customer_key, link user to the appropriate enterprise.
+
+        If the key is not found, returns 400
+        If the ket is not valid, return 422
+        """
+        try:
+            enterprise_customer_key_match = models.EnterpriseCustomerInviteKey.objects.get(uuid=pk)
+        except models.EnterpriseCustomerInviteKey.DoesNotExist:
+            return Response({"error": "Could not find Enterprise Customer Invite Key"}, status=HTTP_400_BAD_REQUEST)
```

Comment:
Nit: I think a 404 is more appropriate if the object can't be found.

Reply:
This makes sense to me!

Reply:
There is a `get_object_or_404` method you might use here.